gooddata-sdk/gooddata_sdk/catalog/data_source/model/data_source.py

```diff
@@ -1,25 +1,117 @@
 # (C) 2022 GoodData Corporation
 from __future__ import annotations
 
-from gooddata_sdk.catalog.entity import CatalogEntity
+import base64
+from dataclasses import dataclass
+from typing import Any, List, Optional, Tuple
 
+from gooddata_metadata_client.model.json_api_data_source_in import JsonApiDataSourceIn
+from gooddata_metadata_client.model.json_api_data_source_in_attributes import JsonApiDataSourceInAttributes
+from gooddata_metadata_client.model.json_api_data_source_in_document import JsonApiDataSourceInDocument
+from gooddata_sdk.catalog.entity import CatalogNameEntity
 
-class CatalogDataSource(CatalogEntity):
-    @property
-    def name(self) -> str:
-        return self._e["name"]
 
-    @property
-    def data_source_type(self) -> str:
-        return self._e["type"]
+@dataclass
+class CatalogDataSource(CatalogNameEntity):
```

Comment:
Add custom method __repr__. There is work with password and token, which is set by default to None, but user can set it to values of password and token, then it will be printed with other variables.

Reply:
supressed from repr by field customization